### PR TITLE
Support for ClickHouse array types (e.g. [1,2,3])

### PR DESCRIPTION
**Pull request by monadbobo**

### CODE REVIEW

File: tests/sqpparser_clickhouse.rs [hunk 1]
```diff
@@ -40,16 +41,32 @@ fn parse_map_access_expr() {
...
-            })]
+                distinct: false,
+            })],
         },
         expr_from_projection(only(&select.projection)),
     );
 }
 
+#[test]
+fn parse_array_expr() {
```
Comment (alamb) [inline tests/sqpparser_clickhouse.rs:55]: Can you please also add a test for the `named` variant.

I think you added support for `SELECT ARRAY['1', '2'])` as well but it isn't covered in the tests

**Review by alamb — COMMENTED** [review]
Thank you for the contribution @monadbobo

**Comment by alamb** [comment]
FYI I hope to make the 0.15.0 release in the next day or two, so hopefully we can get this PR in for that release;

File: tests/sqpparser_clickhouse.rs (hunk 1 above)
Comment (monadbobo) [reply tests/sqpparser_clickhouse.rs:55]: named array has been tested in sqlparser_postgres.rs, so I didn't test named array separately here。

Comment (alamb) [reply tests/sqpparser_clickhouse.rs:55]: Found it https://github.com/sqlparser-rs/sqlparser-rs/blob/1da49c1/tests%2Fsqlparser_postgres.rs#L826-L843 👍 

Thanks

**Comment by alamb** [comment]
🤔  this change seems to have broken a downstream test in DataFusion : https://github.com/apache/arrow-datafusion/pull/1956

In particular, sqlparser now rejects

```sql
SELECT array(c1, cast(c2 as varchar)) FROM test
```

As invalid (in this case `array` is a function name). I'll see if I can get a fix

**Pull request merged**